src/test/static/class/class-memberof.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * @memberof MyNamespace
+ */
+class Class {
+}
+
+/**
+ * @function
+ * @memberof MyNamespace
```

Comment:
According to http://usejsdoc.org/tags-memberof.html it looks like we should be doing `@memberof MyNamespace.Class`

Reply:
ah! thats what i wasn't sure about. 

i wasn't sure if the `@memberof NS.Class` form was being used only when the associated function isn't defined on the class.

e.g.:

```ts
/** @memberof Namespace */
MyClass.Foo; // becomes Namespace.MyClass.Foo

/** @memberof Namespace.MyClass */
function Foo() { } // somewhere else we add this to MyClass, so we have to tell it the class in memberof, not only the ns. becomes Namespace.MyClass.Foo
```

Reply:
@justinfagnani i just ran this through `jsdoc --explain`, i think `Namespace` alone is right but we should also be annotating with `@instance` in polymer otherwise jsdoc detects `scope` as `static`.

so more like:

```ts
/**
 * @memberof Polymer
 * @instance
 * @return {void}
 * @function
 */
MyClass.prototype.MyMethod;
```

however in the analyser we probably don't need to care about the existence of `@instance`. rather that is just something we should include in our annotations.

Reply:
yes this is confirmed.

* `@memberof MyName` becomes `MyName.MyClass#MyMethod`
* `@memberof MyName.MyClass` becomes `MyName.MyClass.MyClass#MyMethod`

Reply:
Weird. That's fine I guess, but can you document it all with links where helpful?

Reply:
sure. i guess it behaves this way because the expression already contains namespace information. whereas in the case of a lone function, it doesn't have any clue. combining an already-namespaced name with a namespace = the duplicate you saw.

Reply:
This is pretty confusing!

I spent some time playing around with different forms to see how `jsdoc` would react. The forms that say `GOOD` below render as I would expect in the generated HTML documentation (as a non-static method of `MyNamespace.Class`).

`M1` is the simplest working form that should already work in Analyzer (without this PR), where we just fully qualify the declaration instead of using the local class name. Why don't we do this inside `polymer.dom.html` when declaring the `DomApi` methods (which is the motivation for this PR)?

`M2` is interesting because apparently `jsdoc` is smart enough to follow that `M2` is part of `Class` is part of `MyNamespace` without any annotation hints.

`M3` is the simplest form I could find that has explicit annotations. Note that the `@alias` is required for `jsdoc` (see `M7` for the version without `@alias`).

```js
/**
 * @namespace
 */
var MyNamespace;

/**
 * @memberof MyNamespace
 */
class Class {}

/**
 * GOOD
 *
 * @function
 */
MyNamespace.Class.prototype.M1;

/**
 * GOOD
 *
 * @function
 */
Class.prototype.M2;

/**
 * GOOD
 *
 * @function
 * @memberof MyNamespace.Class.prototype
 * @alias M3
 */
Class.prototype.M3;

/**
 * GOOD
 *
 * @function
 * @memberof! MyNamespace.Class.prototype
 * @alias M4
 */
Class.prototype.M4;

/**
 * BAD: On MyNamespace page as "(static) Class#5"
 *
 * @function
 * @memberof MyNamespace
 */
Class.prototype.M5;

/**
 * BAD: On Class page as "(static) Class#M6"
 *
 * @function
 * @memberof MyNamespace.Class
 */
Class.prototype.M6;

/**
 * BAD: Missing entirely.
 *
 * @function
 * @memberof MyNamespace.Class.prototype
 */
Class.prototype.M7;

/**
 * BAD: On Class page as "Class#M8"
 *
 * @function
 * @memberof! MyNamespace.Class.prototype
 */
Class.prototype.M8;
```

Reply:
thanks so much @aomarks  for investigating this. its all rather confusing and not-so-well-documented.

M2 is what we had originally but doesn't work for us because the analyser doesn't know what namespace it belongs to (it doesn't have any knowledge of the `DomApi` class its self at the point it detects these methods).

I did notice the `static` problem of M5 too. adding `@instance` to resolve it ended up with `Namespace#Class#Method` which is nonsense.

I got confused because m5 did end up correct almost (`namespace.class#method` but was marked as static). i guess i was missing `@alias` when i had tried it with the full name.